crypto/dh/dh_key.c

```diff
@@ -56,11 +56,23 @@ static DH_METHOD dh_ossl = {
     NULL
 };
 
+static const DH_METHOD *default_DH_method = &dh_ossl;
+
 const DH_METHOD *DH_OpenSSL(void)
 {
     return &dh_ossl;
 }
 
+void DH_set_default_method(const DH_METHOD *meth)
```

Comment:
This isn't thread safe. Either we should put locks around it, or just document it as a non-thread safe function. The same for the other similar functions below.